homeassistant/components/enocean/sensor.py

```diff
@@ -32,31 +32,43 @@
 SENSOR_TYPE_POWER = "powersensor"
 SENSOR_TYPE_TEMPERATURE = "temperature"
 SENSOR_TYPE_WINDOWHANDLE = "windowhandle"
+SENSOR_TYPE_DOOR_WINDOW = "door"
 
 SENSOR_TYPES = {
     SENSOR_TYPE_HUMIDITY: {
         "name": "Humidity",
         "unit": "%",
         "icon": "mdi:water-percent",
         "class": DEVICE_CLASS_HUMIDITY,
+        "orig_state": None,
     },
     SENSOR_TYPE_POWER: {
         "name": "Power",
         "unit": POWER_WATT,
         "icon": "mdi:power-plug",
         "class": DEVICE_CLASS_POWER,
+        "orig_state": None,
     },
     SENSOR_TYPE_TEMPERATURE: {
         "name": "Temperature",
         "unit": TEMP_CELSIUS,
         "icon": "mdi:thermometer",
         "class": DEVICE_CLASS_TEMPERATURE,
+        "orig_state": None,
     },
     SENSOR_TYPE_WINDOWHANDLE: {
         "name": "WindowHandle",
         "unit": None,
         "icon": "mdi:window",
         "class": None,
+        "orig_state": None,
+    },
+    SENSOR_TYPE_DOOR_WINDOW: {
```

Comment:
How many states does this sensor have?

Reply:
The WINDOWHANDLE type or the one I added DOOR_WINDOW ?
The WINDOWHANDLE has various I think but I don't have the details (like opened, closed, inclination...)
The new DOOR_WINDOW one has two : Opened and Closed.

Regards.

Reply:
I'm referring to `SENSOR_TYPE_DOOR_WINDOW`. If it only has two states it should just be a binary sensor and part of the binary sensor platform.

Reply:
I thought about that but had problems because binary sensors has no real state as someone told me in the Discord. 
In my tests, I would open the door for instance, the sensor would go to "opened" state, but it would go back to the original state after a few seconds, and the person told me it was normal, that I had to create another "binary_sensor" in HA that would have to be updated based on automation fired by the binary sensor event of enocean...
As I understood, the enocean binary sensor is just an event, like a push of a button that instantly goes back to "not pressed" after releasing.

Reply:
That's the problem with the current design of the binary sensor platform in this integration. If the device is state less and just fires events, it should not be a binary sensor entity. It should just be a custom home assistant event.

Reply:
I understand... So between this and the other integration "design rules" problem, I should just remove the PR, no ?
But is anyone working on the integration's rewrite ? Because with these changes at least other devices work in HA... But I understand your point that if people keep adding stuff instead of rewriting the integration, you think nobody will ever do it, right ?

Reply:
I'm not sure how much would be left of this PR with this limitation.

I don't believe anyone is currently working on this integration.

Yes, unfortunately that's how we have act to avoid getting even further away from our architecture when we have legacy design.

Reply:
ok that's a shame, so I'll just keep on using my local changes to enjoy my devices :-p